odoo/addons/base/models/res_partner.py

```diff
@@ -145,11 +139,11 @@ def _get_default_color(self):
 
     name = fields.Char('Name', required=True, translate=True)
     color = fields.Integer(string='Color', default=_get_default_color, aggregator=False)
-    parent_id: ResPartnerCategory = fields.Many2one('res.partner.category', string='Category', index=True, ondelete='cascade')
-    child_ids: ResPartnerCategory = fields.One2many('res.partner.category', 'parent_id', string='Child Tags')
+    parent_id = fields.Many2one['ResPartnerCategory'](string='Category', index=True, ondelete='cascade')
```

Comment:
Don't we need to import (in TYPE_CHECKING) classes for relational field typing? Or at least indicate `'base.ResPartnerCategory'`?
Otherwise, how will the static analysis know where that type is defined.

Something like this may be needed if you don't use any type information:
```python
from odoo.addons import base  # noqa: F401
```

Reply:
For consistency, we can write base but the class is defined above. Since we are using a ClassName to modelName conversion, this is not necessary, and moreover typing works since there is the class in the file.

Reply:
Ok, I missed it. So it works ok.